app/src/main/java/com/duckduckgo/app/settings/NewSettingsViewModel.kt

```diff
@@ -0,0 +1,303 @@
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchAboutScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchAccessibilitySettings
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchAddHomeScreenWidget
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchAppTPOnboarding
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchAppTPTrackersScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchAppearanceScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchAutofillSettings
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchCookiePopupProtectionScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchDefaultBrowser
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchEmailProtection
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchEmailProtectionNotSupported
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchFireButtonScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchGeneralSettingsScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchMacOs
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchPermissionsScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchPrivateSearchWebPage
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchSyncSettings
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchWebTrackingProtectionScreen
+import com.duckduckgo.app.settings.NewSettingsViewModel.Command.LaunchWindows
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.autoconsent.api.Autoconsent
+import com.duckduckgo.autofill.api.AutofillCapabilityChecker
+import com.duckduckgo.autofill.api.email.EmailManager
+import com.duckduckgo.common.utils.ConflatedJob
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.duckplayer.api.DuckPlayer
+import com.duckduckgo.duckplayer.api.DuckPlayer.DuckPlayerState.DISABLED_WIH_HELP_LINK
+import com.duckduckgo.duckplayer.api.DuckPlayer.DuckPlayerState.ENABLED
+import com.duckduckgo.mobile.android.app.tracking.AppTrackingProtection
+import com.duckduckgo.subscriptions.api.Subscriptions
+import com.duckduckgo.sync.api.DeviceSyncState
+import javax.inject.Inject
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+@SuppressLint("NoLifecycleObserver")
+@ContributesViewModel(ActivityScope::class)
+class NewSettingsViewModel @Inject constructor(
```

Comment:
same here, this is inheriting all the bad practices we’ve been adding over the years. It’s going to take you much longer than if you start the screen from scratch with the new design.